### PR TITLE
Add support for RHEL 8 to GCP capi provider image builder

### PR DESCRIPTION
**Pull request by kahun**
What this PR does / why we need it:
Adds support to build rhel 8 images to using with CAPG provider using cluster-api.

### CODE REVIEW

**Comment by jsturtevant** [comment]
the failure was centos which should be addressed in https://github.com/kubernetes-sigs/image-builder/pull/1135
/retest

File: images/capi/packer/gce/packer.json [hunk 1]
```diff
@@ -6,17 +6,17 @@
...
         "build_timestamp": "{{user `build_timestamp`}}",
-        "distribution": "ubuntu",
+        "distribution": "{{user `distribution`}}",
         "distribution_release": "{{user `distribution_release`}}",
         "distribution_version": "{{user `distribution_version`}}",
         "kubernetes_version": "{{user `kubernetes_semver` | clean_resource_name}}"
       },
       "machine_type": "{{ user `machine_type` }}",
-      "name": "{{user `build_name`}}",
+      "name": "{{ user `build_name` }}",
```
Comment (mboersma) [inline images/capi/packer/gce/packer.json:15]: This is a nitpick, but if we're trying to be consistent in templates, I think we should omit these spaces. That's how we have it already for the majority of template variables, and that's how HashiCorp has it [in their docs](https://developer.hashicorp.com/packer/docs/templates/legacy_json_templates/user-variables#usage).

**Review by mboersma — COMMENTED** [review]
/lgtm

File: images/capi/packer/gce/packer.json (hunk 1 above)
Comment (kahun) [reply images/capi/packer/gce/packer.json]: I've removed the spaces & also an unnecesary symlink

**Pull request merged**